back/src/data/source/time-traveller.db.datasource.ts

```diff
@@ -26,10 +26,17 @@ export class TimeTravellerDataSource {
     })
   }
 
-  save(timeTraveller: TimeTravellerModel & { password: string; salt: string }) {
-    return this.timeTravellerRepository.save({
-      ...timeTraveller,
-      last_login_at: new Date()
-    })
+  loginUpset(timeTraveller: TimeTravellerModel & { password: string; salt: string })
```

Comment:
typo `loginUpset` -> `loginUpsert`
```suggestion
  loginUpsert(timeTraveller: TimeTravellerModel & { password: string; salt: string })
```